test/jdk/javax/print/PrintServiceLookup/CountPrintServices.java

```diff
@@ -25,33 +25,35 @@
 import java.io.InputStreamReader;
 import javax.print.PrintService;
 import javax.print.PrintServiceLookup;
-import javax.print.attribute.AttributeSet;
-import javax.print.attribute.HashAttributeSet;
-import javax.print.attribute.standard.PrinterName;
+
+import jtreg.SkippedException;
 
 /*
  * @test
  * @bug 8032693
  * @key printer
+ * @library /test/lib/
+ * @requires (os.family == "linux")
  * @summary Test that lpstat and JDK agree whether there are printers.
  */
 public class CountPrintServices {
 
   public static void main(String[] args) throws Exception {
-    String os = System.getProperty("os.name").toLowerCase();
-    System.out.println("OS is " + os);
-    if (!os.equals("linux")) {
-        System.out.println("Linux specific test. No need to continue");
-        return;
-    }
     PrintService services[] =
         PrintServiceLookup.lookupPrintServices(null, null);
     if (services.length > 0) {
        System.out.println("Services found. No need to test further.");
        return;
     }
     String[] lpcmd = { "lpstat", "-a" };
-    Process proc = Runtime.getRuntime().exec(lpcmd);
+    Process proc = null;
+    try {
+        proc = Runtime.getRuntime().exec(lpcmd);
+    } catch (java.io.IOException e) {
+        if (e.getMessage().contains("No such file or directory")) {
+            throw new SkippedException("Cannot find lpstat");
+        }
+    }
     proc.waitFor();
```

Comment:
If `IOException` is thrown, `proc` will be `null` which will produce `NullPointerException`.

I think you should re-throw the `IOException` if its message is different from the case where you throw `SkippedException`.

Reply:
I have re-thow the IOException if the SkippedException do not throw.

But I have no idea how to verify this special situation( `Runtime.getRuntime().exec(lpcmd)` throw IOException but do not contains "No such file or directory" stack trace )